src/MassTransit/Context/SendContextProxy.cs

```diff
@@ -7,7 +7,7 @@ namespace MassTransit.Context
 
     public class SendContextProxy<TMessage> :
         SendContextProxy,
-        PublishContext<TMessage>

```

Comment:
SendContext/PublishContext are the same type, thus the reason for a single set of proxies.

Reply:
was trying to do that on my prev project, didn't work

Reply:
Hmm. It should, I'd have to verify but the changes to send are copied in publish by default.

Reply:
It might just be topology:https://github.com/MassTransit/MassTransit/blob/develop/src/MassTransit/Transports/InMemory/Configuration/Configuration/InMemoryTopologyConfiguration.cs#L44

Reply:
let me know if it works or fixed, I'll remove for publish

Reply:
I don't think it replicates for send/publish, but it's a single type so I'd think a single provider could be used that works for either pipeline. To avoid duplication.

Reply:
I've removed publish

Reply:
@phatboyg https://github.com/MassTransit/MassTransit/pull/1815/files#diff-a04709b87c5ad209d478232b88ca641bR26
you can check this test, for publish it will fail, so it seems I was right or misconfigured something